File: go/AtbashCipher.go
package main

import (
	"fmt"
	"strings"
	"bufio"
	"os"
)

func atbashCipher(plainText string) string {
	result := []rune{}
	var c rune = 0

	for _, char := range plainText {
		if char >= 97 && char <= 122 {
			c = (25 - (char - 97)) % 26 + 97

			result = append(result, c)
		} else {
			result = append(result, char)
		}
	}

	return string(result)
}

func main() {
	fmt.Println("Enter string to encode/decode:")
	
	plainText, _ := bufio.NewReader(os.Stdin).ReadString('\n')

	result := atbashCipher(strings.ToLower(plainText))

	fmt.Printf("Result: %v", result)
}

File: php/atbash-cipher.php
<?php
declare(strict_types=1);

function atbashCipher(string $text): string {
  $result = "";
  $char = 0;

  for ($i = 0; $i < strlen($text); $i++) {
    $code = ord($text[$i]);

    if ($code >= 97 && $code <= 122) {
      $char = (25 - (ord($text[$i]) - 97)) % 26 + 97;
      $result .= chr($char);
    } else {
      $result .= $text[$i];
    }
  }

  return $result;
}

echo "Enter text to encode/decode: \n";

$text = trim(fgets(fopen("php://stdin", "r")));

$text = strtolower($text);

$result = atbashCipher($text);

echo "Result: $result";